### ISSUE TITLE
`tests/ui/treat-err-as-bug/panic-causes-oom-112708.rs` is very slow after #121927

### ISSUE THREAD

**Issue by Zalathar** [rust-lang/rust#122009]
```rust
//@ compile-flags: -Ztreat-err-as-bug
//@ dont-check-failure-status
//@ error-pattern: aborting due to `-Z treat-err-as-bug=1`
//@ dont-check-compiler-stderr
//@ rustc-env:RUST_BACKTRACE=0

fn main() {
    #[deny(while_true)]
    while true {}
}
```

When doing `x test ui`, I noticed that this particular test now takes a very long time to run. I was able to bisect the change down to #121927.

```text
# BEFORE
$ git checkout cd9e5b5f && x test ui -- panic-causes-oom --force-rerun
...
test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 16251 filtered out; finished in 26.35ms

# AFTER
$ git checkout de95c39a && x test ui -- panic-causes-oom --force-rerun
...
test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 16251 filtered out; finished in 103.41s
```

(Notice the jump from *milliseconds* to *seconds*.)

@rustbot label +A-testsuite +A-contributor-roadblock +T-compiler

**Comment by Zalathar** [comment]
cc @Zoxc

Any idea what's happening here?

**Comment by Zoxc** [comment]
It's running out of memory. It will be fixed by https://github.com/rust-lang/rust/pull/121993.

**Comment by Zalathar** [comment]
Fixed by #121993.
